src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -0,0 +1,366 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Definition;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Models;
+using CompartmentType = Microsoft.Health.Fhir.ValueSets.CompartmentType;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Everything
+{
+    public class PatientEverythingService : IPatientEverythingService
+    {
+        private readonly Func<IScoped<ISearchService>> _searchServiceFactory;
+        private readonly ISearchOptionsFactory _searchOptionsFactory;
+        private readonly ISearchParameterDefinitionManager _searchParameterDefinitionManager;
+        private readonly ICompartmentDefinitionManager _compartmentDefinitionManager;
+
+        private IReadOnlyList<string> _includes = new[] { "general-practitioner", "organization" };
```

Comment:
>"general-practitioner", "organization" [](http://example.com/codeflow?start=58&length=38)

what about `link`?

Reply:
We do not search `link` here, because including `link` might make the result Bundle containing duplicate resources. For example, if Patient A and B link to each other in `link`, when doing [base]\Patient\A\$everything, Patient B would be returned twice, one pulled by _include and the other pulled by Patient A's compartment search. By discussing with Caitlin, returning duplicate resources is a deal breaker, so we do not include `link` in `_include`.